### PR TITLE
#119 Authentication failure messages

### LINKED ISSUE #119: Authentication failure messages
### Version

- Python Version: [e.g. 3.8.6]

### Describe the bug

- When signing in, if incorrect password is entered, no message is shown and the sign in page is reloaded.
- When signing in, if non-existing email address is used for username, a message is shown saying the email is incorrect. This exposes signed up emails.

### Steps to reproduce

1. Go to `/account/signin`
2. Sign in using a correct email and incorrect password / using incorrect email.

### Expected behavior

Unsuccessful signing in show return a message without giving any information about reason of failure. 

### Actual behavior

As explained above.

### Screenshots

N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by fadi-albasha**
- Added a message to show sign in failure when the password is incorrect.
- Made the message shown when email is not signed up similar to the message when the password is incorrect.
- Removed validation error messages in the sign in page.

close #119

### CODE REVIEW

File: app/controllers/user_controller.py [hunk 1]
```diff
@@ -39,3 +39,30 @@ def get_following(_, username: str) -> Any:
...
+
+@user_routes.route('/unfollow', methods=['POST'])
+@login_required
+def unfollow_user(user_id: int) -> Any:
+    user_id_to_remove: int = request.form.values()
+    if user_id_to_remove == user_id:
+        return '', 400
+    try:
+        User.remove_from_following(user_id, user_id_to_remove)
+        return '', 200
```
Comment (jason-gerard) [inline app/controllers/user_controller.py]: This should be 204, success with no response payload

**Review by jason-gerard — COMMENTED** [review]
Other than that looks good

File: app/controllers/user_controller.py (hunk 1 above)
Comment (fadi-albasha) [reply app/controllers/user_controller.py]: I know, I tried to change it to 204 in this commit but it resulted in a conflict. I'll fix it later.

Comment (fadi-albasha) [reply app/controllers/user_controller.py]: It's fixed now.

**Review by jason-gerard — DISMISSED** [review]
Fixed the conflict, when that happens its because your branch is behind main just have to rebase

**Review by mnashat — DISMISSED** [review]

**Review by LujainKhalaf — CHANGES_REQUESTED** [review]
Since this is a sign in page, I believe the best error message would be: Incorrect username and/OR password.
Having all of them is too excessive for an error message, and since it's just a sign-in page, the length of the password required was demanded during the the sign up, so that can be excluded from error sign in, same thing for the valid email.

**Comment by fadi-albasha** [comment]
I agree with @LujainKhalaf, I'll remove validation error messages form the sign in page.

**Comment by fadi-albasha** [comment]
I removed validation error messages in the sign in page.

**Review by mnashat — APPROVED** [review]

**Review by LujainKhalaf — APPROVED** [review]
Great work!

**Pull request merged**